lib/trino-filesystem/src/main/java/io/trino/filesystem/fileio/BulkOperationsFileIo.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.fileio;
+
+import io.trino.filesystem.TrinoFileSystem;
+import org.apache.iceberg.io.BulkDeletionFailureException;
+import org.apache.iceberg.io.SupportsBulkOperations;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulkOperationsFileIo
+        extends ForwardingFileIo
+        implements SupportsBulkOperations
+{
+    private static final int DELETE_BATCH_SIZE = 1000;
+
+    public BulkOperationsFileIo(TrinoFileSystem fileSystem)
+    {
+        super(fileSystem);
+    }
+
+    @Override
+    public void deleteFiles(Iterable<String> pathsToDelete)
+            throws BulkDeletionFailureException
+    {
+        List<String> filesToDelete = new ArrayList<>();
+        pathsToDelete.forEach(path -> {
+            filesToDelete.add(path);
+            if (filesToDelete.size() == DELETE_BATCH_SIZE) {
+                deleteBatch(filesToDelete);
+                filesToDelete.clear();
+            }
+        });
+        if (!filesToDelete.isEmpty()) {
+            deleteBatch(filesToDelete);
+        }
+    }
+
+    private void deleteBatch(List<String> filesToDelete)
+    {
+        try {
+            fileSystem.deleteFiles(filesToDelete);
+        }
+        catch (IOException e) {
+            // TODO find out the number of files that failed to delete and throw BulkDeletionFailureException
```

Comment:
what is `org.apache.iceberg.io.BulkDeletionFailureException#numberFailedObjects` needed for?

if we don't need it, we can skip having a TODO

Reply:
> SupportsBulkOperations::deleteFiles throws BulkDeletionFailureException exception.

I know. 
BTW it can also throw any unckeched exception, obviously.

> numberFailedObjects which means the number of files that failed to delete.

I figured this from the name :)
How does Iceberg lib use that information?